swift/Arrow/Sources/Arrow/ArrowDecoder.swift

```diff
@@ -105,6 +125,11 @@ public class ArrowDecoder: Decoder {
         return array.asAny(self.rbIndex) as? T
     }
 
+    func doDecodeSingleValue<T>() throws -> T {
+        let array: AnyArray = try self.getCol(self.singleRBCol)
+        return array.asAny(self.rbIndex) as! T // swiftlint:disable:this force_cast
```

Comment:
Can we call `doDecode<T>(_ col: Int)` instead of adding a similar code?

Reply:
Will do.  My goal was to make it clear that the index for the single column was being handled a bit different but sending in the singleRBCol to the doDecode method works.

Reply:
Ah, defining `doDecodeSingleValue()` and `isNullSingleValue()` is OK. I just wanted to say that changing the implementation something like:

```swift
    func doDecodeSingleValue<T>() throws -> T? {
        return doDecode(self.singleRBCol)
    }
```

Reply:
Aha, I see!  Thanks for clarifying!  We do remove the overhead of another func call by removing the SingleValue methods.  Let me know what you think.

Reply:
If there is a performance overhead, I'm OK with the current approach.